lib/init.coffee

```diff
@@ -10,6 +15,82 @@ module.exports =
       type: 'string'
       default: ''
 
-
   activate: ->
-    console.log 'activate linter-erlc'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-erlc.executablePath',
+    (executablePath) =>
+      @executablePath = executablePath
+  deactivate: ->
+    @subscriptions.dispose()
+  provideLinter: ->
+    helpers = require('atom-linter')
+    provider =
+      name: 'Erlc'
+      grammarScopes: ['source.erlang']
+      scope: 'file'
+      lintOnFly: false
+      lint: (textEditor) =>
+        filePath = textEditor.getPath()
+        args = ["-Wall", "-v", "-o", os.tmpDir()]
+
+        pa = atom.config.get 'linter-erlc.pa'
+        for path in pa.split(",").map((x) -> x.trim()).filter((x) -> x)
+          args.push("-pa")
+          args.push(path)
+
+        projectPath = null
+        if path.basename(path.dirname(filePath)) == 'src'
+          projectPath = path.dirname(path.dirname(filePath))
+
+        includeDirs = atom.config.get 'linter-erlc.includeDirs'
+        for includePath in includeDirs.split(",").map((x) -> x.trim()).filter((x) -> x)
+          args.push("-I")
+          if projectPath && !includePath.match(/^\//)
+            args.push(path.join(projectPath, includePath))
+          else
+            args.push(includePath)
+
+        args.push(filePath)
+
+
+        return helpers.exec(@executablePath, args).then (result) ->
+          toReturn = []
+          patterns = [
+            {regex: /(.+):(\d+):\s(.+)/, cb: (m) -> {lineNo: m[2], text: m[3]}},
+            {regex: /(.+): (.+)/, cb: (m) -> {lineNo: 1, text: m[2]}}
+          ]
+          lines = result.split('\n')
+          sourceLines = textEditor.getText().split("\n")
```

Comment:
Use `textEditor.getBuffer().getLines()` (see [here](https://atom.io/docs/api/v1.4.0/TextBuffer#instance-getLines)) to get the lines with their line endings _properly_ stripped, using the above you will have dangling `\r` on the lines if the file was originally `CRLF`.